ui/redux/actions/comments.js

```diff
@@ -573,12 +580,35 @@ export function doCommentCreate(
   return async (dispatch: Dispatch, getState: GetState) => {
     const state = getState();
     const activeChannelClaim = selectActiveChannelClaim(state);
+    let mentionedChannels: Array<MentionedChannel> = [];
 
     if (!activeChannelClaim) {
       console.error('Unable to create comment. No activeChannel is set.'); // eslint-disable-line
       return;
     }
 
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll
+    // $FlowFixMe
+    const mentionMatches = [...comment.matchAll(MENTION_REGEX)];
+
+    if (mentionMatches.length > 0) {
+      mentionMatches.forEach((match) => {
+        const matchTerm = match[0];
+        const mention = matchTerm.substring(matchTerm.indexOf('@'));
+        const mentionUri = `lbry://${mention}`;
+
+        const claim = selectClaimForUri(state, mentionUri);
+
+        if (claim) {
+          mentionedChannels.push({ channelName: claim.name, channelID: claim.claim_id });
+        } else if (claim === undefined) {
+          dispatch(doResolveUri(mentionUri))
```

Comment:
yeah I thought this way would be safer because there's many cases where it could never get resolved like typing fast before the debounce timing for search, or copy pasting a mention, typing less characters than lighthouse minimum, etc